py/src/braintrust/wrappers/genai/__init__.py

```diff
@@ -0,0 +1,443 @@
+
+        # known limitations with today's converters
+        clean_kwargs["model"] = input["model"]
+
+        start = time.time()
+        with start_span(
+            name="generate_content", type=SpanTypeAttribute.LLM, input=input, metadata=clean_kwargs
+        ) as span:
+            result = wrapped(*args, **kwargs)
+            metrics = _extract_generate_content_metrics(result, start)
+            span.log(output=result, metrics=metrics)
+            return result
+
+    wrap_function_wrapper(Models, "_generate_content", wrap_generate_content)
+
+    def wrap_generate_content_stream(wrapped: Any, instance: Any, args: Any, kwargs: Any):
+        input, clean_kwargs = get_args_kwargs(args, kwargs, ["model", "contents", "config"])
+
+        input = _serialize_input(instance._api_client, input)
+
+        # known limitations with today's converters
+        clean_kwargs["model"] = input["model"]
+
+        start = time.time()
+        first_token_time = None
+        with start_span(
+            name="generate_content_stream", type=SpanTypeAttribute.LLM, input=input, metadata=clean_kwargs
+        ) as span:
+            chunks = []
+            for chunk in wrapped(*args, **kwargs):
+                if first_token_time is None:
+                    first_token_time = time.time()
+                chunks.append(chunk)
+                yield chunk
+
+            aggregated, metrics = _aggregate_generate_content_chunks(chunks, start, first_token_time)
+            span.log(output=aggregated, metrics=metrics)
+            return aggregated
+
+    wrap_function_wrapper(Models, "generate_content_stream", wrap_generate_content_stream)
+
+    mark_patched(Models)
+    return Models
+
+
+def wrap_async_models(AsyncModels: Any):
+    if is_patched(AsyncModels):
+        return AsyncModels
+
+    async def wrap_generate_content(wrapped: Any, instance: Any, args: Any, kwargs: Any):
+        input, clean_kwargs = get_args_kwargs(args, kwargs, ["model", "contents", "config"])
+
+        input = _serialize_input(instance._api_client, input)
+
+        # known limitations with today's converters
+        clean_kwargs["model"] = input["model"]
+
+        start = time.time()
+        with start_span(
+            name="generate_content", type=SpanTypeAttribute.LLM, input=input, metadata=clean_kwargs
+        ) as span:
+            result = await wrapped(*args, **kwargs)
+            metrics = _extract_generate_content_metrics(result, start)
+            span.log(output=result, metrics=metrics)
+            return result
+
+    wrap_function_wrapper(AsyncModels, "generate_content", wrap_generate_content)
+
+    async def wrap_generate_content_stream(wrapped: Any, instance: Any, args: Any, kwargs: Any):
+        input, clean_kwargs = get_args_kwargs(args, kwargs, ["model", "contents", "config"])
+
+        input = _serialize_input(instance._api_client, input)
+
+        # known limitations with today's converters
```

Comment:
this comment is kinda confusing to me. doesn't say much

Reply:
I'll go ahead and remove. I left it more like a breadcrumb for future us, but not significant.